lib/server.js

```diff
@@ -1309,13 +1309,19 @@ Server.prototype._routeErrorResponse = function _routeErrorResponse(
             return;
         }
 
-        // Expose only knonw errors
-        if (_.isNumber(err.statusCode)) {
+        // only automatically send errors that are known (e.g., restify-errors)
+        if (err && _.isNumber(err.statusCode)) {
             res.send(err);
             return;
         }
 
-        res.send(new errors.InternalError(emitErr || err));
+        var finalErr = emitErr || err;
+        if (!(finalErr instanceof Error)) {
+            // if user land domain threw a string or number or any other
+            // non-error object, handle it here as best we can.
+            finalErr = (finalErr && finalErr.toString()) || '';
```

Comment:
How does this behave if `finalErr` is `false` or `0`? Do we expect both of those to return `''`